packages/discord.js/src/util/Transformers.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+const snakeCase = require('lodash.snakecase');
+const transform = require('lodash.transform');
+
+class Transformers extends null {
+  /**
+   * Transforms camel-cased keys into snake cased keys
+   * @param {*} obj The object to transform
+   * @returns {*}
+   */
+  static toSnakeCase(obj = {}) {
+    return transform(obj, (acc, value, key, target) => {
+      const camelKey = Array.isArray(target) ? key : snakeCase(key);
+      acc[camelKey] = Boolean(value) && typeof value === 'object' ? Transformers.toSnakeCase(value) : value;
```

Comment:
why Boolean(value)?

Reply:
Because `null`'s `typeof` is `object`

Reply:
> Because `null`'s `typeof` is `object`

a ternary is an if, an if turns everything into a boolean already